integration-cli/docker_cli_events_test.go

```diff
@@ -446,6 +446,15 @@ func (s *DockerSuite) TestEventsFilterContainer(c *check.C) {
 	}
 }
 
+func (s *DockerSuite) TestEventsInvalidFilterType(c *check.C) {
+	since := daemonUnixTime(c)
+	dockerCmd(c, "run", "--rm", "busybox", "true")
+	out, _, err := dockerCmdWithError("events", "--since", since, "--filter", "nonexisttype=aaa")
+	c.Assert(err, check.NotNil)
+	expected := "Error response from daemon: Invalid filter 'nonexisttype'"
+	c.Assert(out, checker.Contains, expected, check.Commentf("Missing filter validation error from daemon"))
+}
```

Comment:
I don't think this needs to be an integration test. I think a unit test should cover it.